rows.go

```diff
@@ -438,6 +436,11 @@ func CollectRows[T any](rows Rows, fn RowToFunc[T]) ([]T, error) {
 	return slice, nil
 }
 
+// CollectRows iterates through rows, calling fn for each row, and collecting the results into a slice of T.
+func CollectRows[T any](rows Rows, fn RowToFunc[T]) ([]T, error) {
+	return AppendRows([]T(nil), rows, fn)
```

Comment:
After updating from 5.5.1 to 5.5.3, our integration tests failed.

```
Diff:
--- Expected
+++ Actual
@@ -12,4 +12,3 @@
unknownFields: ([]uint8) <nil>,
- Edges: ([]*edgev1.Edge) {
- }
+ Edges: ([]*edgev1.Edge) <nil>
})
```

When `SELECT * FROM table WHERE ...` returns no rows, 5.5.1 returns empty slice:

https://github.com/jackc/pgx/blob/ba05097642a802eb5a76343f27a9a3515ed86498/rows.go#L424

5.5.3 returns `nil` slice:

https://github.com/jackc/pgx/blob/6f8f6ede6c67ce6b755feafe5c44f93577545147/rows.go#L441

@jackc I believe this is an unintentional change introduced in this PR.

Reply:
I don't think that `CollectRows` was ever documented to return the dangling slice instead of the nil slice when no items were collected, but it's a simple enough fix (just change the call to `AppendRows` to be `AppendRows([]T{}, rows, fn)`).

Reply:
I don't think empty vs. nil slice was promised, but may as well preserve the previous behavior.